backend/interface.go

```diff
@@ -14,6 +17,28 @@ const (
 	Config        = "config"
 )
 
+func ParseType(s string) (Type, error) {
+	if s == string(Data) {
+		return Data, nil
+	}
+	if s == string(Key) {
+		return Key, nil
+	}
+	if s == string(Lock) {
+		return Lock, nil
+	}
+	if s == string(Snapshot) {
+		return Snapshot, nil
+	}
+	if s == string(Index) {
+		return Index, nil
+	}
+	if s == string(Config) {
+		return Config, nil
+	}
+	return "", errors.New("invalid type")
```

Comment:
If you care about style and look, this could be shortened to:

```Go
	switch s {
		case string(Data): return Data, nil
		case string(Key): return Data, nil
		case string(Lock): return Lock, nil
		case string(Snapshot): return Snapshot, nil
		case string(Index): return Index, nil
		case string(Config): return Config, nil
	}
	return "", errors.New("invalid type")
```